wandb/util.py

```diff
@@ -307,6 +310,146 @@ def get_h5_typename(o):
+    ):
+        return local_file_uri_to_path(path_or_uri)
+    else:
+        return None
+
+
+def make_tarfile(output_filename, source_dir, archive_name, custom_filter=None):
+    # Helper for filtering out modification timestamps
+    def _filter_timestamps(tar_info):
+        tar_info.mtime = 0
+        return tar_info if custom_filter is None else custom_filter(tar_info)
+
+    unzipped_filename = tempfile.mktemp()
+    try:
+        with tarfile.open(unzipped_filename, "w") as tar:
+            tar.add(source_dir, arcname=archive_name, filter=_filter_timestamps)
+        # When gzipping the tar, don't include the tar's filename or modification time in the
+        # zipped archive (see https://docs.python.org/3/library/gzip.html#gzip.GzipFile)
+        with gzip.GzipFile(
+            filename="", fileobj=open(output_filename, "wb"), mode="wb", mtime=0
+        ) as gzipped_tar, open(unzipped_filename, "rb") as tar:
+            gzipped_tar.write(tar.read())
+    finally:
+        os.remove(unzipped_filename)
+
+
+def _user_args_to_dict(arguments):
+    user_dict = {}
+    i = 0
+    while i < len(arguments):
+        arg = arguments[i]
+        split = arg.split("=", maxsplit=1)
+        # flag arguments don't require a value -> set to True if specified
+        if len(split) == 1 and (
+            i + 1 >= len(arguments) or arguments[i + 1].startswith("-")
+        ):
+            name = split[0].lstrip("-")
+            value = True
+            i += 1
+        elif len(split) == 1 and not arguments[i + 1].startswith("-"):
+            name = split[0].lstrip("-")
+            value = arguments[i + 1]
+            i += 2
+        elif len(split) == 2:
+            name = split[0].lstrip("-")
+            value = split[1]
+            i += 1
+        if name in user_dict:
+            wandb.termerror("Repeated parameter: '%s'" % name)
+            sys.exit(1)
+        user_dict[name] = value
+    return user_dict
+
+
+def exec_cmd(
+    cmd,
+    throw_on_error=True,
+    env=None,
+    stream_output=False,
+    cwd=None,
+    cmd_stdin=None,
+    **kwargs
+):
+    """
+    Runs a command as a child process.
+    A convenience wrapper for running a command from a Python script.
+    Keyword arguments:
+    cmd -- the command to run, as a list of strings
+    throw_on_error -- if true, raises an Exception if the exit code of the program is nonzero
+    env -- additional environment variables to be defined when running the child process
+    cwd -- working directory for child process
+    stream_output -- if true, does not capture standard output and error; if false, captures these
+      streams and returns them
+    cmd_stdin -- if specified, passes the specified string as stdin to the child process.
+    Note on the return value: If stream_output is true, then only the exit code is returned. If
+    stream_output is false, then a tuple of the exit code, standard output and standard error is
+    returned.
```

Comment:
We should lint this docstring.

Reply:
This is actually unused at this time. I thought I had removed this, but I guess I just moved it. Removing it